deploy/samples/subscriptions.yaml

```diff
@@ -6,4 +6,5 @@ metadata:
 spec:
   sql: SELECT "quantity", "product_id" AS KEY FROM INVENTORY."products_on_hand"
   database: RAWKAFKA
-
+  hints:
+    kafka.numPartitions: "2"
```

Comment:
It is unclear whether this hint is for the Hoptimator to get more information about the source, or it is a hint towards the destination. This is important especially, that the User is submitting the select * query, without any mention of the destination/destination type. If this information is hard-coded, or inferred from hints, then hints keywords should be more contextual.

Reply:
The Kafka topic controller will try to apply `kafka.numPartitions` to all Kafka topics used by a pipeline, since the hint is the same throughout the entire pipeline. This means an existing Kafka topic could get re-partitioned even if it is only used as a source. This sorta makes sense, since end-to-end pipelines probably _should_ be partitioned all the same. And if the controller cannot honor a hint, it will just give up. Repartitioning an existing Kafka topic could just fail and we move on.

My proposed logic is: we try to honor hints across the entire pipeline. For source topics that we don't own, we won't have permission to add partitions, which is fine. For source topics that we do own (e.g. topics that are sinks in some other subscription), we can go ahead and add partitions. That's probably the least-confusing behavior.

With #33, it's possible for adapters to have different templates (and thus, different hints) for reading vs writing. This gives us a way around the default behavior for cases where it doesn't make sense. For example, we could ignore `numPartitions` when reading from a Kafka topic, and enforce it when writing to a Kafka topic.